incubator/exporter-prometheus/src/main/java/io/aklivity/zilla/runtime/exporter/prometheus/internal/utils/ObjectObjectObjectFunction.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2021-2022 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.exporter.prometheus.internal.utils;
+
+@FunctionalInterface
+public interface ObjectObjectObjectFunction<R, T, U, V>
```

Comment:
This is most likely better named as a `TriFunction` similar to `BiFunction`.
We do specify the types in the functional interface name when primitives are involved, to determine their placement among the type vars.